lib/cylc/scheduler.py

```diff
@@ -1166,12 +959,35 @@ def load_suiterc(self, reconfigure):
         printcfg(self.config.cfg, handle=handle)
         handle.close()
 
+    def _load_initial_cycle_point(self, _, row):
+        """Load previous initial cycle point.
+
+        For restart, it may be missing from "suite.rc", but was specified as a
+        command line argument on cold/warm start.
+        """
+        key, value = row
+        if key == "initial_point":
+            self._cli_initial_point_string = value
+            self.do_process_tasks = True
+
     def configure_suite(self, reconfigure=False):
         """Load and process the suite definition."""
 
-        if self.is_restart:
-            self._cli_initial_point_string = self.get_state_file_info()[0]
-            self.do_process_tasks = True
+        if reconfigure:
+            self.pri_dao.take_checkpoints("reload-init", [self.pub_dao])
```

Comment:
May be nice to use `other_daos=[self.pub_dao]` - reads naively as 'take the checkpoints using only `self.pub_dao`'